packages/block-editor/src/components/element-button/index.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import RichText from '../rich-text';
+
+const getElementProps = ( props ) => {
+	const newProps = { ...props };
+	newProps.className += ' wp-element-button';
+	return newProps;
+};
+
+export default class ElementButton extends Component {
```

Comment:
This should be a function component, there's not reason to use a class.

Reply:
When I use a function component I get this error:

```react-dom.js?ver=6.0-beta2-53224-src:61 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?```

I assume that's because the Button block passes a ref down to this component. Is it worth trying to change that approach first?

Reply:
Oh I see, I think the problem is that you're trying to absorb "RichText" in this component while you could do this when you need RichText

```
<RichText tagName={ ElementButton } />
```

That said, I see that you're forcing ElementButton to be a `button` tag while we at least have two use-cases: an `a` for the case of the button block and a `button` in the case of the search block. Are these the only two use cases for this element?

If that's the case we can consider having two "components" `ElementButtonAsLink` `ElementButtonAsButton` or something like that.

Reply:
Indeed it's a great question that would benefit from a broader perspective. I think it does personally, we had a [lengthy discussion](https://github.com/WordPress/gutenberg/issues/7534) about it in the context of the `Button` component of Gutenberg as it's also a component that can render both as a link or a "button" depending on its props. 

As long as we stick with these two tags (a, button), I think it's fine.

Reply:
(That's said about the "ref" error, I think the component should forwardRef anyway :) )